### ISSUE TITLE
Kitsune proxy panics

### ISSUE THREAD

**Issue by mattyg** [holochain-apps/relay#246]
User reported this issue via telelgram:

> Here is the terminal output of the DesktopApp under Kubuntu 22.04:
```
WEBKIT_DISABLE_COMPOSITING_MODE=1 volla_messages
# lair-keystore connection_url # unix:///home/tuxus/.local/share/volla-messages/holochain/0.7/keystore/socket?k=z0Nvc-2elyl-ahdOEsOt_WE6FncxMCVM3oCaZFj3BlA #
# lair-keystore running #
FATAL PANIC PanicInfo {
    payload: Any { .. },
    message: Some(
        called `Result::unwrap()` on an `Err` value: InvalidLastSymbol(5, 115),
    ),
    location: Location {
        file: "/home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/kitsune_p2p_proxy-0.4.0-rc.0/src/proxy_url.rs",
        line: 147,
        col: 84,
    },
    can_unwind: true,
    force_no_backtrace: false,
}
thread 'tokio-runtime-workerFATAL PANIC ' panicked at PanicInfo/home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/kitsune_p2p_proxy-0.4.0-rc.0/src/proxy_url.rs {
:    147payload: Any { .. }:,
84    :
called `Result::unwrap()` on an `Err` value: InvalidLastSymbol(5, 115)message
: note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Some(
        called `Result::unwrap()` on an `Err` value: InvalidLastSymbol(5, 115),
    ),
    location: Location {
        file: "/home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/kitsune_p2p_proxy-0.4.0-rc.0/src/proxy_url.rs",
        line: 147,
        col: 84,
    },
    can_unwind: true,
    force_no_backtrace: false,
}
thread 'tokio-runtime-worker' panicked at /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/kitsune_p2p_proxy-0.4.0-rc.0/src/proxy_url.rs:147:FATAL PANIC 84:
called `Result::unwrap()` on an `Err` value: InvalidLastSymbol(5, 115)
PanicInfo {
    payload: Any { .. },
    message: Some(
        called `Result::unwrap()` on an `Err` value: InvalidLastSymbol(5, 115),
    ),
    location: Location {
        file: "/home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/kitsune_p2p_proxy-0.4.0-rc.0/src/proxy_url.rs",
        line: 147,
        col: 84,
    },
    can_unwind: true,
    force_no_backtrace: false,
}
FATAL PANIC PanicInfo {
    payload: Any { .. },
    message: Some(
        called `Result::unwrap()` on an `Err` value: InvalidLastSymbol(5, 115),
    ),
    location: Location {
        file: "/home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/kitsune_p2p_proxy-0.4.0-rc.0/src/proxy_url.rs",
        line: 147,
        col: 84,
    },
    can_unwind: true,
    force_no_backtrace: false,
}
thread 'tokio-runtime-worker' panicked at /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/kitsune_p2p_proxy-0.4.0-rc.0/src/proxy_url.rs:147:84:
called `Result::unwrap()` on an `Err` value: InvalidLastSymbol(5, 115)
thread 'tokio-runtime-worker' panicked at /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/kitsune_p2p_proxy-0.4.0-rc.0/src/proxy_url.rs:147:84:
called `Result::unwrap()` on an `Err` value: InvalidLastSymbol(5, 115)
```

**Comment by mattyg** [comment]
It looks like the build of 0.7.1 I prematurely shared to the telegram testers group was accidentally built off `develop`, and thus is running holochain 0.4-rc and so the network is incompatible :facepalm:

Still I don't understand why this panic occurs. Need to investigate further.

**Comment by mattyg** [comment]
The panic has been resolved in holochain develop-0.4 by https://github.com/holochain/holochain/issues/4170
